apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -562,25 +563,35 @@ API.v1.addRoute(
 
 API.v1.addRoute(
 	'channels.close',
-	{ authRequired: true },
+	{ authRequired: true, validateParams: isChannelsCloseProps },
 	{
 		async post() {
-			const findResult = await findChannelByIdOrName({
-				params: this.bodyParams,
-				checkedArchived: false,
+			const receivedRoomId = 'roomId' in this.bodyParams ? this.bodyParams.roomId : undefined;
+			const receivedRoomName = 'roomName' in this.bodyParams ? this.bodyParams.roomName : undefined;
```

Comment:
```suggestion
			const receivedRoomId = this.bodyParams.roomId;
			const receivedRoomName = this.bodyParams.roomName;
```
Won't this have the same effect? :thinking:

Reply:
This approach gives a type error such as `roomId does not exist on {roomName: string}` 🙁

Reply:
Ohhhhh, that's because you're using `oneOf` in the rest-typing definition
Try using `anyOf` instead. This will allow both params to be sent together
Sorry for the wrong suggestion there :confused: